lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -23,253 +23,308 @@
 #include "swift/AST/ReferencedNameTracker.h"
 #include "swift/AST/Types.h"
 #include "swift/Frontend/FrontendOptions.h"
+#include "swift/Frontend/ReferenceDependencyKeys.h"
 #include "swift/Basic/LLVM.h"
 #include "llvm/ADT/MapVector.h"
 #include "llvm/ADT/SetVector.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/YAMLParser.h"
 
+// See docs/DependencyAnalysis.rst.
+
 using namespace swift;
 
-static void findNominalsAndOperators(
-    llvm::MapVector<const NominalTypeDecl *, bool> &foundNominals,
-    llvm::SmallVectorImpl<const FuncDecl *> &foundOperators,
-    DeclRange members) {
-  for (const Decl *D : members) {
-    auto *VD = dyn_cast<ValueDecl>(D);
-    if (!VD)
-      continue;
+std::vector<std::string>
+swift::reversePathSortedFilenames(const ArrayRef<std::string> elts) {
+  std::vector<std::string> tmp(elts.begin(), elts.end());
+  std::sort(tmp.begin(), tmp.end(),
+            [](const std::string &a, const std::string &b) -> bool {
+              return std::lexicographical_compare(a.rbegin(), a.rend(),
+                                                  b.rbegin(), b.rend());
+            });
+  return tmp;
+}
 
-    if (VD->hasAccess() &&
-        VD->getFormalAccess() <= AccessLevel::FilePrivate) {
-      continue;
-    }
+using namespace reference_dependency_keys;
```

Comment:
Nitpick: Please keep all the top-level `using namespace` declarations together.

Reply:
Done!